File: article/src/main/java/com/ruzhan/awaker/article/util/KeyboardUtils.java
package com.ruzhan.awaker.article.util;

import android.content.Context;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

/**
 * Copyright ©2017 by ruzhan
 */

public final class KeyboardUtils {

    private KeyboardUtils() {
    }


    public static void openSoftInput(Context context, View editText) {
        InputMethodManager inputMethodManager =
                (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
        inputMethodManager.showSoftInput(editText, 0);
    }

    public static void closeSoftInput(Context context, View editText) {
        InputMethodManager inputMethodManager =
                (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
        inputMethodManager.hideSoftInputFromWindow(editText.getWindowToken(), 0);
    }
}

File: common/src/main/java/com/ruzhan/common/util/KeyboardUtils.kt
package com.ruzhan.common.util

import android.content.Context
import android.view.View
import android.view.inputmethod.InputMethodManager

object KeyboardUtils {

    @JvmStatic
    fun openSoftInput(context: Context, editText: View) {
        val inputMethodManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        inputMethodManager.showSoftInput(editText, 0)
    }

    @JvmStatic
    fun closeSoftInput(context: Context, editText: View) {
        val inputMethodManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        inputMethodManager.hideSoftInputFromWindow(editText.windowToken, 0)
    }
}
